### PR TITLE
Issue1050 media choices

### LINKED ISSUE #1050: Provide easier to understand choice when selecting a media
I observed that (new) users often have problems selecting a media. This is not surprising, as the default Medium list looks like this:
screenshot at 2018-11-09 14-16-31

I suggest we constrain the media that show up in the dialog box to these:
screenshot at 2018-11-09 14-16-00

This can be done with the declaration
```modelica
  replaceable package Medium =
    Modelica.Media.Interfaces.PartialMedium "Medium in the component"
      annotation (choices(
        choice(redeclare package Medium = IBPSA.Media.Air "Moist air"),
        choice(redeclare package Medium = IBPSA.Media.Water "Water"),
        choice(redeclare package Medium = IBPSA.Media.Antifreeze.PropyleneGlycolWater(
          property_T=293.15,
          X_a=0.40)
          "Propylene glycol water")));
```
This ticket is to make such an implementation. I suggest we don't include the media in `IBPSA.Media.Specialized` as these are in my view of little use to almost all users. Advanced users who really have a need for such media can still enter them manually rather than through the pull down menu.

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #1050.

### CODE REVIEW

**Comment by mwetter** [comment]
@JayHuLBL This is not ready to merge, please see the broken tests.

**Comment by JayHuLBL** [comment]
@mwetter It's ready for review when the test passed.
Updated the media choice as suggested, except the one in `IBPSA.Fluid.Sources.BaseClasses.Outside`, `IBPSA.Fluid.Sources.MassFlowSource_WeatherData`, `IBPSA.Fluid.Sources.MassFlowSource_T`, `IBPSA.Fluid.Sources.TraceSubstancesFlowSource` where other media is necessary.

**Comment by mwetter** [comment]
@JayHuLBL : Go through the pull request and revert those changes that are not needed. For example, `IBPSA/Airflow/Multizone/Examples/OneOpenDoor.mo` and other files have changes that are not needed. Only commit and put into the pull request intended changes.

**Comment by mwetter** [comment]
@Mathadon Can you please review and merge after the holidays so that we have a non-LBL person in the development-review process. Thanks.

**Comment by mwetter** [comment]
@JayHuLBL : The model `Outside` can be configured to use glycol or water, yet it only makes sense to use it with air. Please correct it.
Also, why is the new base class `PartialSource_volume` needed, and what does the `_volume` means to say as there is no control volume in this model. Can you use the existing `PartialSource` base class instead?

File: IBPSA/Fluid/Sources/BaseClasses/PartialSource_volume.mo [hunk 1]
```diff
@@ -0,0 +1,65 @@
+within IBPSA.Fluid.Sources.BaseClasses;
+partial model PartialSource_volume
+  "Partial component source with one fluid connector"
+    import Modelica.Constants;
```
Comment (Mathadon) [inline IBPSA/Fluid/Sources/BaseClasses/PartialSource_volume.mo]: don't we try to avoid import statements?

File: IBPSA/Fluid/Sources/BaseClasses/PartialSource_volume.mo [hunk 2]
```diff
@@ -0,0 +1,65 @@
+within IBPSA.Fluid.Sources.BaseClasses;
+partial model PartialSource_volume
+  "Partial component source with one fluid connector"
```
Comment (Mathadon) [inline IBPSA/Fluid/Sources/BaseClasses/PartialSource_volume.mo]: It would be good if the comment specifies how this component differs from the regular `PartialSource`

File: IBPSA/Airflow/Multizone/BaseClasses/TwoWayFlowElement.mo [hunk 3]
```diff
@@ -9,8 +9,16 @@ partial model TwoWayFlowElement "Flow resistance that uses the power law"
...
   extends IBPSA.Airflow.Multizone.BaseClasses.ErrorControl;
 
-  replaceable package Medium = Modelica.Media.Interfaces.PartialMedium
-    annotation (choicesAllMatching=true);
+  replaceable package Medium =
+    Modelica.Media.Interfaces.PartialMedium "Medium in the component"
+      annotation (choices(
+        choice(redeclare package Medium = IBPSA.Media.Air "Moist air"),
+        choice(redeclare package Medium = IBPSA.Media.Water "Water"),
+        choice(redeclare package Medium =
```
Comment (Mathadon) [inline IBPSA/Airflow/Multizone/BaseClasses/TwoWayFlowElement.mo]: We could add an air medium that includes CO2?

Comment (mwetter) [reply IBPSA/Airflow/Multizone/BaseClasses/TwoWayFlowElement.mo]: @Mathadon : I suggest to keep it at these 3 mediums for now, and add it if we see a clear need.

**Comment by mwetter** [comment]
@JayHuLBL : Can you please address the remaining comments, in particular remove `PartialSource_volume.mo` unless it is really needed.

**Comment by JayHuLBL** [comment]
@mwetter @Mathadon 
It's ready for review.

File: IBPSA/Resources/Scripts/Dymola/Fluid/Sources/Examples/MassFlowSource_WeatherData.mos [hunk 4]
```diff
@@ -1,3 +1,4 @@
+Advanced.StoreProtectedVariables = true;
```
Comment (mwetter) [inline IBPSA/Resources/Scripts/Dymola/Fluid/Sources/Examples/MassFlowSource_WeatherData.mos]: Why is this needed, and if it is needed, why don't you set it back to its old value after the simulation and plotting?

File: IBPSA/Airflow/Multizone/BaseClasses/TwoWayFlowElement.mo (hunk 3 above)
Comment (mwetter) [inline IBPSA/Airflow/Multizone/BaseClasses/TwoWayFlowElement.mo]: @JayHuLBL : Why do you allow the multizone airflow models to be configured with water or glycol? I can't see a use case where this makes sense.

File: IBPSA/Fluid/Actuators/Dampers/MixingBox.mo [hunk 5]
```diff
@@ -3,8 +3,15 @@ model MixingBox "Outside air mixing box with interlocked air dampers"
...
-    annotation (choicesAllMatching = true);
+    Modelica.Media.Interfaces.PartialMedium "Medium in the component"
+      annotation (choices(
+        choice(redeclare package Medium = IBPSA.Media.Air "Moist air"),
+        choice(redeclare package Medium = IBPSA.Media.Water "Water"),
+        choice(redeclare package Medium =
+            IBPSA.Media.Antifreeze.PropyleneGlycolWater (
+          property_T=293.15,
+          X_a=0.40)
+          "Propylene glycol water, 40% mass fraction")));
```
Comment (mwetter) [inline IBPSA/Fluid/Actuators/Dampers/MixingBox.mo]: @JayHuLBL : Why using an air damper with water or glycol? Please change. This could even be set as default to moist air.

File: IBPSA/Fluid/FMI/Adaptors/ThermalZone.mo [hunk 6]
```diff
@@ -2,8 +2,16 @@ within IBPSA.Fluid.FMI.Adaptors;
...
+  replaceable package Medium =
+    Modelica.Media.Interfaces.PartialMedium "Medium in the component"
+      annotation (choices(
+        choice(redeclare package Medium = IBPSA.Media.Air "Moist air"),
+        choice(redeclare package Medium = IBPSA.Media.Water "Water"),
+        choice(redeclare package Medium =
+            IBPSA.Media.Antifreeze.PropyleneGlycolWater (
+          property_T=293.15,
+          X_a=0.40)
+          "Propylene glycol water, 40% mass fraction")));
```
Comment (mwetter) [inline IBPSA/Fluid/FMI/Adaptors/ThermalZone.mo]: @JayHuLBL : A thermal zone with water or glycol as the medium?

File: IBPSA/Fluid/FMI/BaseClasses/X_w_toX.mo [hunk 7]
```diff
@@ -3,8 +3,15 @@ block X_w_toX "Conversion from Xi to X"
...
-     annotation (choicesAllMatching=true);
+    Modelica.Media.Interfaces.PartialMedium "Medium in the component"
+      annotation (choices(
+        choice(redeclare package Medium = IBPSA.Media.Air "Moist air"),
+        choice(redeclare package Medium = IBPSA.Media.Water "Water"),
+        choice(redeclare package Medium =
+            IBPSA.Media.Antifreeze.PropyleneGlycolWater (
+          property_T=293.15,
+          X_a=0.40)
+          "Propylene glycol water, 40% mass fraction")));
```
Comment (mwetter) [inline IBPSA/Fluid/FMI/BaseClasses/X_w_toX.mo]: @JayHuLBL : This certainly makes no sense to be configured for water or glycol.

File: IBPSA/Airflow/Multizone/BaseClasses/TwoWayFlowElement.mo (hunk 3 above)
Comment (JayHuLBL) [reply IBPSA/Airflow/Multizone/BaseClasses/TwoWayFlowElement.mo]: @mwetter I will further change the settings, as well as other inappropriate changes.

File: IBPSA/Resources/Scripts/Dymola/Fluid/Sources/Examples/MassFlowSource_WeatherData.mos (hunk 4 above)
Comment (JayHuLBL) [reply IBPSA/Resources/Scripts/Dymola/Fluid/Sources/Examples/MassFlowSource_WeatherData.mos]: The class `sin_with_h` extends `IBPSA.Fluid.Sources.BaseClasses.PartialSource`, in which `medium` is protected.

**Comment by mwetter** [comment]
@Mathadon : Are you OK with the changes, in particular the refactoring https://github.com/ibpsa/modelica-ibpsa/pull/1052/commits/5f7eeec41d6a478fc4409511c1f6a7254254382a

**Comment by Mathadon** [comment]
@mwetter the refactoring looks good!
